### ISSUE TITLE
unify json production as much as possible

### ISSUE THREAD

**Issue by gleckler1** [PCMDI/pcmdi_metrics#415]
@doutriaux1 @lee1043   CD can provide example in one of a number of existing branches (e.g, https://github.com/PCMDI/pcmdi_metrics/blob/334_monsoon_wang/src/python/devel/monsoon_wang/scripts/monsoon-wang_driver.py) and JL and PG can propagate across other branches

**Comment by gleckler1** [comment]
@doutriaux1 @lee1043  PG using monsoon-wang as a test case using JSON class... we will need to improve and apply this in all other analysis codes

**Comment by doutriaux1** [comment]
@gleckler1 I think we can close this one now. Correct?

**Comment by gleckler1** [comment]
@doutriaux1 Yes, but to advance the solution we need to have the PMPparser as part of the PMP master on github so that we can efficiently advance the common/default arguments (https://github.com/PCMDI/pcmdi_metrics/wiki/PMPParser)

**Comment by doutriaux1** [comment]
@zshaheen I think we can merge this in today, correct? @zshaheen isn't the Parser already in?

**Comment by zshaheen** [comment]
@doutriaux1 yes, the parser is in, but not under a main release on conda. So you have to download a nightly to use it.

**Comment by gleckler1** [comment]
@zshaheen @doutriaux1  We need the parser **in the PMP github master.**   Can you do a pull request for just the parser, NOT the rest of the CDP?

**Comment by zshaheen** [comment]
But it is in master. See here: https://github.com/PCMDI/pcmdi_metrics/blob/master/src/python/pcmdi/pmp_parser.py

I think you want it where a user doesn't need to install CDP, is that correct?

**Comment by doutriaux1** [comment]
@zshaheen I think @gleckler1 didn't realize he could already use it in his branch

**Comment by doutriaux1** [comment]
@zshaheen can you point @gleckler1 to your wiki page on how to use this for his driver

**Comment by gleckler1** [comment]
@zshaheen @doutriaux1 OK, I think I'm getting it.  Its in master, but only in the nightly conda build (not the v1.1 release).   Do I have the process (below) right for making changes?

From a nigthly conda env, I go to ~/git/pcmdi_metrics, create a branch to modify the default CDP parameters in (~git/pcmdi_metrics/src/python/pcmdi/pmp_parser.py), push it, make a pull request, and once that gets accepted/merged into master the next day the changes will be available in the nightly conda build of the PMP.     

Is that right??

**Comment by doutriaux1** [comment]
it should be, assuming @zshaheen to get the automatic nighlty build is in as well.

**Comment by zshaheen** [comment]
@gleckler1 Here's the page on how to use the parser, which I think you already have: https://github.com/PCMDI/pcmdi_metrics/wiki/PMPParser

This is how the driver will work, once we accept the pull request (after modifications to account for the changes after late November): https://github.com/PCMDI/pcmdi_metrics/wiki/PMP-using-CDP-Guide

@gleckler1 yes, your statement is correct. 

@doutriaux1 I guess I'll try to get the nightly builds working today.

**Comment by gleckler1** [comment]
@doutriaux1 @zshaheen  Thanks...  using the parser rnow (e.g., https://github.com/PCMDI/pcmdi_metrics/blob/334_monsoon_wang/src/python/devel/monsoon_wang/scripts/mpi_compute.py)...  finessing and augmenting the default options will be how we work towards unifying different metrics.

**Comment by zshaheen** [comment]
@doutriaux1 @gleckler1 Nightly uploads are now working. It'll only upload the pull requests when they get accepted into master, not before.
